test/communication_test.go

```diff
@@ -37,7 +37,8 @@ func TestMain(m *testing.M) {
 	go server.Start(7008)
 
 	// Establish a connection to previously started server
-	testGrpcClientConn, err = client.GetConnection(testGrpcServerEndpoint, "testdata/server.pem")
+	testGrpcClientConn, err = client.GetConnection(testGrpcServerEndpoint,
+		"testdata/server.pem", true)
```

Comment:
Set `insecure` to `false` here since you're passing the cert anyway.

Reply:
Also, could you add another test where you don't specify the cert (use empty string), and set `insecure` to `true`?

Reply:
Done and done.